Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Fleet/Driver.hs

```diff
@@ -1034,20 +1034,47 @@ getDriverFleetStatus :: ShortId DM.Merchant -> Context.City -> Text -> Maybe Tex
 getDriverFleetStatus _merchantShortId _opCity requestorId mbFleetOwnerId = do
   requestedPerson <- QPerson.findById (Id requestorId) >>= fromMaybeM (PersonDoesNotExist requestorId)
   (entityRole, entityId) <- validateRequestorRoleAndGetEntityId requestedPerson mbFleetOwnerId
-  getDriverFlowStatusCounts entityId entityRole
-  where
-    getDriverFlowStatusCounts :: Text -> DP.Role -> Flow Common.DriverStatusRes
-    getDriverFlowStatusCounts entityId entityRole = do
+  let statusList = [DDF.ONLINE, DDF.OFFLINE, DDF.SILENT, DDF.ON_PICKUP, DDF.ON_RIDE, DDF.ACTIVE, DDF.INACTIVE]
+      statusKey status = DDF.getStatusKey status entityId
+      allKeys = map statusKey statusList
+  logTagInfo "DriverStatus" $ "Checking Redis for keys: " <> show allKeys <> ", entityRole: " <> show entityRole <> ", entityId: " <> entityId
+  redisCounts <-
+    mapM
+      ( \key ->
+          Redis.get @T.Text key >>= \v -> do
+            logTagInfo "DriverStatus" $ "Redis.get " <> key <> " => " <> show v
+            pure $ v >>= (readInt . T.unpack)
+      )
+      allKeys
+  if all isJust redisCounts
+    then do
+      logTagInfo "DriverStatus" $ "Cache hit for all statuses for entityId: " <> entityId <> ", counts: " <> show redisCounts
+      pure $ toDriverStatusRes (zip (map Just statusList) (map (fromMaybe 0) redisCounts))
+    else do
+      logTagInfo "DriverStatus" $ "Cache miss for some statuses for entityId: " <> entityId <> ". Falling back to ClickHouse. Redis counts: " <> show redisCounts
       driverIds <- case entityRole of
         DP.FLEET_OWNER -> CFDV.getDriverIdsByFleetOwnerId entityId
         DP.OPERATOR -> CDOA.getDriverIdsByOperatorId entityId
         _ -> throwError (InvalidRequest "Invalid Data")
+      logTagInfo "DriverStatus" $ "Fetched driverIds from association: " <> show driverIds
       driverModeInfo <- CDI.getModeCountsByDriverIds driverIds
+      logTagInfo "DriverStatus" $ "ClickHouse returned: " <> show driverModeInfo <> ". Updating Redis."
+      forM_ driverModeInfo $ \(mbStatus, count) ->
+        case mbStatus of
+          Just status -> do
+            let k = statusKey status
+            logTagInfo "DriverStatus" $ "Redis.set " <> k <> " = " <> show count
+            Redis.set @T.Text k (T.pack (show count))
+          Nothing -> pure ()
       pure $ toDriverStatusRes driverModeInfo
-
+  where
+    readInt :: String -> Maybe Int
+    readInt s = case reads s of
+      [(i, _)] -> Just i
+      _ -> Nothing
```

Comment:
why simple `readMaybe` function did not used? Moreover redis can get and set as `Int`:
```
Redis.get @Int key
Redis.set @Int count
```